cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowUtil.qll

```diff
@@ -318,3 +318,14 @@ VariableAccess getAnAccessToAssignedVariable(Expr assign) {
     result = var.getAnAccess()
   )
 }
+
+/** A guard that validates some expression. */
+class BarrierGuard extends Expr {
+  /** Holds if this guard validates `e` upon evaluating to `branch`. */
+  abstract predicate checks(Expr e, boolean branch);
```

Comment:
Can you mark this predicate as `deprecated` without getting any warnings in library code? In any case, please write in the QLDoc for C++/IR/C# that this feature is not yet supported.